### ISSUE TITLE
Creating style from history with multiple edits, applies multiple instances of edited modules

### ISSUE THREAD

**Issue by nyanpasu64** [darktable-org/darktable#17225]
### Describe the bug

If I edit an effect so it appears multiple times in an image's history, then create a style from the lighttable view, the style will duplicate the module when applied.

### Steps to reproduce

1. Import an image.
2. Adjust its exposure.
3. Switch to lighttable view, on the right expand the "styles" manager and click "create..."
	- There will be two or more instances of "exposure", all checked by default.
4. Type a name and click "save".
5. Optionally discard history on this image.
6. With mode set to either append or overwrite, apply the style to the same image or another.

The exposure effect will be applied twice, once with the initial settings and once with the new altered settings.

This gets worse if you apply a style in "append" mode over an existing image, then create a new style without "compress history stack" first. In this case, *all* modules are duplicated (including a ridiculous-looking double lens correction).

### Expected behavior

Darktable should uncheck or hide modules which have been overwritten.

### Logfile | Screenshot | Screencast

- Log: [darktable-log.txt](https://github.com/user-attachments/files/16424519/darktable-log.txt)

4.8.1 "create new style" dialog with bug (two exposure effects):
![darktable_r1KSoEwM6K](https://github.com/user-attachments/assets/a35c8f69-6972-48b4-94c1-67cffa1b8333)

4.6.1 dialog with no bug (first exposure effect is omitted):
![darktable_bVg4AG2Hdy](https://github.com/user-attachments/assets/248575aa-fc3a-463d-8e4e-c6e590ec78b9)

### Commit

45e83f8679eded6aebe796a25138e6971edf3c1d most likely

### Where did you obtain darktable from?

downloaded from www.darktable.org

### darktable version

4.8.1

### What OS are you using?

Windows

### What is the version of your OS?

Windows 11 23H2

### Describe your system?

Ryzen 5 5600X, 16 GB RAM.

### Are you using OpenCL GPU in darktable?

Yes

### If yes, what is the GPU card and driver?

RX 570 4GB, Adrenalin 24.3.1

### Please provide additional context if applicable. You can attach files too, but might need to rename to .txt or .zip

- On 4.8.0 and 4.8.1, the bug is reproducible after fresh edit, on clean profile, etc.
- Style file with two exposure modules: [2tmp.dtstyle.txt](https://github.com/user-attachments/files/16424439/2tmp.dtstyle.txt)
- Darktable 4.6.1 is unaffected; it hides overwritten history elements from the "create..." dialog, but includes duplicates of a module.

**Comment by nyanpasu64** [comment]
Bug still happens on nightly 4.9.0+621~gf737cc4fdd.

![darktable_IjZZIcFdi0](https://github.com/user-attachments/assets/0f7065e9-55d0-4c2d-9774-64fed63ae3ad)

**Comment by nyanpasu64** [comment]
By default all instances of a module have the "include" column checked. If I create a style without unchecking the overridden ones, applying it elsewhere produces an image with *all* history versions of a module applied as multiple instances (rather than only the currently active revision), which doesn't match the source image (and can produce questionable results like multiple instances of Filmic).

I'd say this is a quite unexpected behavior, and makes it harder to replicate the currently active effects of an image on another (because you have to manually uncheck inactive revisions of an effect when creating a style).

**Comment by Sean-McConnachie** [comment]
I have to agree with @nyanpasu64... creating a new instance of the same style, instead of replacing or overriding the original one completely destroys batch editing workflows.

I'm currently trying to apply a simple exposure correction to a set of astro photos and this issue is making it more difficult - I would consider this a bug from a UX standpoint.

**Comment by Tooniis** [comment]
The image that the style gets create from has one instance of a module, then the style creation dialog gives multiple instances. How is that exactly expected behavior? I expect to have only the module instances that I have in the image such that when I apply the style to a different image it gets the exact same look as the one where the style originated from, while stacking history instances makes it look wildly (and uselessly) different.

**Comment by wpferguson** [comment]
Compress the history stack of the source image before creating the style.  If you are after the "final" look, you don't need all the intermediate steps.

Styles may appear to be simple, but they are not especially when using multiple module instances.  Create your style, then test it to make sure it does what you meant for it to do.
